libs/remote_storage/src/lib.rs

```diff
@@ -48,16 +48,16 @@ const REMOTE_STORAGE_PREFIX_SEPARATOR: char = '/';
 /// The prefix is an implementation detail, that allows representing local paths
 /// as the remote ones, stripping the local storage prefix away.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct RemotePath(PathBuf);
+pub struct RemotePath(Utf8PathBuf);
 
 impl std::fmt::Display for RemotePath {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0.display())
+        write!(f, "{}", self.0.as_std_path().display())
```

Comment:
Utf8Path and Utf8PathBuf `impl std::fmt:Display`, so this could be just:

```suggestion
        std::fmt::Display::fmt(&self.0, f)
```